Shared/UserAgent.swift

```diff
@@ -108,10 +109,12 @@ public struct CustomUserAgentConstant {
         "paypal.com": defaultMobileUA,
         "yahoo.com": defaultMobileUA,
         "disneyplus.com": customDesktopUA,
-        "ecosia.org": ecosiaMobileUA ]
+        URLProvider.production.domain: ecosiaMobileUA,
+        URLProvider.staging.domain: ecosiaMobileUA]
```

Comment:
Great catch indeed! 🔥 
QQ 👇 
As the app instance is sealed within its environment and the interactions with it will be solely handled by the current domain (e.g. either prod or staging), I was wondering whether would it be sufficient to add the current domain only 🤔 
```suggestion
        Environment.current.urlProvider.domain: ecosiaMobileUA]
```

Reply:
I'd say it makes sense that we need both urls as we want our user agent to be set for visiting prod and staging domains. I.e. You use our app center app (which uses staging) you can still visit ecosia.org via the url bar. If the user agent is missing you'll see different behaviour (i.e. incentives to download the app) which might confuse testers. Thus, I'd keep both.